### DISCUSSION TITLE
Too many private field declaration without access methods

### DISCUSSION THREAD

**Discussion by patrickdalla** [jfree/jfreechart#329] [Ideas]
The code is full of private access field declaration. This makes some simple extensions difficult.
For example: I implemented a chart that the value axis starts always on zero, no matter the zoom level. So the zoom painted box should always starts on zero, no matter where I started drawing it. If I had access to zoomPoint, it would be enough to change it to set Y to zero position of the chart. But it is declared as private with no access method. I can implement the correct behaviour accessing zoomPoint with java reflection on JDK 11, but this cannot work on other or future releases of Java.

**Comment by trashgod** [comment]
Instead of accessing `zoomPoint`, consider adding an `AxisChangeListener` that restores the lower bound; do likewise for the domain.

```
rangeAxis.addChangeListener(new AxisChangeListener() {
    @Override
    public void axisChanged(AxisChangeEvent event) {
        EventQueue.invokeLater(() -> {
            rangeAxis.setLowerBound(0);
        });
    }
});
```

**Comment by patrickdalla** [comment]
But I want the zoom rect being drawn when moving the mouse reflect the
final zoom that will be applied if the mouse is released

Em seg., 28 de nov. de 2022 18:56, trashgod ***@***.***>
escreveu:

**Comment by trashgod** [comment]
It seems to work for me using this [example](https://stackoverflow.com/a/41337259/230513).

**Comment by tracylynne99** [comment]
What I did is just modify the appropriate private variables to be protected. True it makes for a bit of a pain in the rear if/when you update your jfreechart baseline, but you'll survive lol.  I do :-)
